kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoKVTable.java

```diff
@@ -185,7 +186,8 @@ public KeyValueIterator<Bytes, byte[]> range(
   public KeyValueIterator<Bytes, byte[]> all(final int kafkaPartition, final long minValidTs) {
     final FindIterable<KVDoc> result = docs.find(Filters.and(
         Filters.not(Filters.exists(KVDoc.TOMBSTONE_TS)),
-        Filters.gte(KVDoc.TIMESTAMP, minValidTs)
+        Filters.gte(KVDoc.TIMESTAMP, minValidTs),
+        Filters.eq(KVDoc.KAFKA_PARTITION, kafkaPartition)
```

Comment:
we should consider adding a configuration that will create an index in MongoDB on the kafka partition

Reply:
#344

Reply:
well we're going to have to get rid of this if we want to support partition scaling. That's why I went with the approach (long ago) of using the partitioner to figure out the kafka partition based on the key

Unfortunately I never found the time to finish that PR, so it's on me 

Well technically the PR was finished but you requested a change that was completely reasonable and that's when I had to pivot to whatever it was that was more important -- point being I can dig that old PR up and finally finish it if we want to do this and it needs to happen ASAP. Thoughts?

Reply:
An aside: how much work is left on that PR @ableegoldman ?